### PR TITLE
feat(pie-cookie-banner): DSW-1702 cookie banner prop bug

### PR DESCRIPTION
**Pull request by kevinrodrigues**
## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: packages/components/pie-cookie-banner/README.md [hunk 1]
```diff
@@ -74,10 +74,11 @@ import { PieCookieBanner } from '@justeattakeaway/pie-cookie-banner/dist/react';
...
-| locale | `Object`  | {English language locale} | Assigns the localisation data for the component strings. |
-| cookieStatementLink | `String`  | `''` | Allows a url to be passed for the cookie statement link. |
-| cookieTechnologiesLink | `String`  | `''` | Allows a url to be passed for the cookie technology link. |
+| hasPrimaryActionsOnly | `Boolean` | `false` | When true, sets the variant to "primary" for the button which accepts necessary cookies only.            |
+| locale | `Object` | {English language locale} | Assigns the localisation data for the component strings.                                                 |
+| cookieStatementLink | `String` | `''` | Allows a url to be passed for the cookie statement link.                                                 |
+| cookieTechnologiesLink | `String` | `''` | Allows a url to be passed for the cookie technology link.                                                |
+| defaultPreferences | `Object` | `''` | Allows { 'functional': true, 'personalized': true, 'analytical': true } to be passed in by the consumer. |
```
Comment (jamieomaguire) [inline packages/components/pie-cookie-banner/README.md:81]: if the type is Object then I don't think default can be an empty string

File: apps/pie-storybook/stories/pie-cookie-banner.stories.ts [hunk 2]
```diff
@@ -66,6 +66,7 @@ const BaseStoryTemplate = (props: CookieBannerProps) : TemplateResult => {
             .cookieStatementLink=${cookieStatementLink}
             .cookieTechnologiesLink=${cookieTechnologiesLink}
             ?hasPrimaryActionsOnly="${hasPrimaryActionsOnly}"
+            .defaultPreferences="${{ 'functional': true, 'personalized': true, 'analytical': true }}"
```
Comment (jamieomaguire) [inline apps/pie-storybook/stories/pie-cookie-banner.stories.ts:69]: NIT: Please store in a variable to keep the story template clean

File: packages/components/pie-cookie-banner/src/defs.ts [hunk 3]
```diff
@@ -62,6 +62,8 @@ export interface CookieBannerProps {
      * Allows consumers to pass in a specific cookie technology link.
      */
     cookieTechnologiesLink: string;
+
+    defaultPreferences: Record<PreferenceIds, boolean> | object
```
Comment (jamieomaguire) [inline packages/components/pie-cookie-banner/src/defs.ts:66]: can any of these flags be omitted? for example could I pass `{ functional: true }` ? If so, I believe the booleans should be boolean | undefined (optional)

File: packages/components/pie-cookie-banner/src/index.ts [hunk 4]
```diff
@@ -206,7 +213,7 @@ export class PieCookieBanner extends LitElement implements CookieBannerProps {
                  </div>
                 <pie-switch
                     id="${id}"
-                    ?checked="${checked}"
+                    ?checked="${this.defaultPreferences[id] || shouldToggleAll || checked}"
```
Comment (jamieomaguire) [inline packages/components/pie-cookie-banner/src/index.ts:216]: Please add some browser tests for the new prop

File: packages/components/pie-cookie-banner/src/defs.ts (hunk 3 above)
Comment (jamieomaguire) [inline packages/components/pie-cookie-banner/src/defs.ts:66]: please add a JSDoc for this prop

File: packages/components/pie-cookie-banner/src/index.ts [hunk 5]
```diff
@@ -197,6 +200,10 @@ export class PieCookieBanner extends LitElement implements CookieBannerProps {
         const descriptionLocaleKey = `preferencesManagement.${id}.description`;
         // Ensure not to display fallback text as description as its expected that some items might not have its own description
         const description = hasDescription && this._localiseText(descriptionLocaleKey);
+        const requiredToggleAllKeys = ['functional', 'personalized', 'analytical'];
```
Comment (jamieomaguire) [inline packages/components/pie-cookie-banner/src/index.ts:203]: I can imagine us having to look back on this in the future, is it worth a comment to explain why these are here?

**Review by jamieomaguire — APPROVED** [review]

**Comment by kevinrodrigues** [comment]
Will follow up @jamieomaguire comments in a separate pr 👍

**Comment by jamieomaguire** [comment]
Feel free to resolve my comments and raise a follow up PR to address them @kevinrodrigues 😃

File: packages/components/pie-cookie-banner/src/index.ts (hunk 5 above)
Comment (jamieomaguire) [inline packages/components/pie-cookie-banner/src/index.ts:203]: @kevinrodrigues I have tested this in Aperture NextJS and it seems like if they are not all true, none of them are enabled. I do not want to hold up this PR but I am concerned this is not correct behaviour.

**Comment by raoufswe** [comment]
merging this but we will need to do a clean up afterwards

**Review by raoufswe — APPROVED** [review]

**Comment by kevinrodrigues** [comment]
Follow up pr to address PR comments: https://github.com/justeattakeaway/pie/pull/1239

**Pull request merged**